src/lib.rs

```diff
@@ -1,21 +1,96 @@
 //! The project that will henceforth be known as `starling`.
-
+//!
+// `error_chain!` can recurse deeply
+#![recursion_limit = "1024"]
 #![deny(missing_docs)]
 #![deny(missing_debug_implementations)]
+#![deny(unsafe_code)]
+// Annoying warning emitted from the `error_chain!` macro.
+#![allow(unused_doc_comment)]
 
+extern crate clap;
+#[macro_use]
+extern crate error_chain;
 extern crate futures;
 extern crate tokio_core;
 
+pub mod cli;
```

Comment:
I really didn't make that clear at all, sorry. I think the use of `cli` shouldn't be in `src/lib.rs`. `src/bin/starling.rs` sounds like a good place for it.